### PR TITLE
Update EU instructions for adding Sendgrid data source

### PR DESCRIPTION
**Pull request by ampcynthli**
# Amplitude Developer Docs PR

## Description
- Add updated text to ensure clarity for EU customers

## Deadline

None

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [x] Doc update.
- [ ] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [ ] My documentation follows the style guidelines of this project.
- [ ] I previewed my documentation on a local server using `mkdocs serve`.
- [ ] Running `mkdocs serve` didn't generate any failures.
- [ ] I have performed a self-review of my own documentation.

@amplitude-dev-docs

### CODE REVIEW

File: docs/data/sources/sendgrid.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ There are no other required setup steps in Amplitude.
 
 1. In SendGrid, find Settings, then click **Mail Settings**.
 2. Select **Event Webhook**.
-3. Under HTTP Post URL, enter the following text: `https://<your-api-key>:CJdAK9fWEn4dH2UzKht37sAM@api.amplitude.com/sendgrid`.
+3. Under HTTP Post URL, enter the following text: `https://<your-api-key>:CJdAK9fWEn4dH2UzKht37sAM@api.amplitude.com/sendgrid`. (If you are in the European Union, ensure you include `.eu` within the URl (i.e. `https://<your-api-key>:CJdAK9fWEn4dH2UzKht37sAM@api.eu.amplitude.com/sendgrid`))
```
Comment (markzegarelli) [inline docs/data/sources/sendgrid.md:25]:
```suggestion
3. Under HTTP Post URL, enter the following text: `https://<your-api-key>:CJdAK9fWEn4dH2UzKht37sAM@api.amplitude.com/sendgrid`. (If you're in the European Union, include `.eu` in the URl (for example `https://<your-api-key>:CJdAK9fWEn4dH2UzKht37sAM@api.eu.amplitude.com/sendgrid`))
```

**Review by markzegarelli — CHANGES_REQUESTED** [review]

**Review by markzegarelli — APPROVED** [review]

File: docs/data/sources/sendgrid.md (hunk 1 above)
Comment (ampcynthli) [reply docs/data/sources/sendgrid.md:25]: No problem; and, URL.

Also, I was hoping to have the new text as a sub-bullet point under #3. I tried to indent it + numbering it as `a.` and `1.` but neither worked. I think that would be a better presentation of this optional note. Thoughts?

**Comment by markzegarelli** [comment]
@ampcynthli Yeah, list formatting is really quirky in this tool (and in Markdown).

I've updated the list so it renders as a sub-bullet, and also broke up the text block by region, for better scanability.

**Comment by ampcynthli** [comment]
@markzegarelli Thank you so much for your help!

**Pull request merged**